### PR TITLE
support apr ranges

### PR DESCRIPTION
**Pull request by 0xSkly**

### CODE REVIEW

**Review by franzns — APPROVED** [review]
Looks good. As daniel requested in the chat, let's add two new fields to the gql type: min and max

Question is, what to do with the total? I'd either set it to 0 or take max value.

**Comment by 0xSkly** [comment]
added the fields and using max in the totals

File: modules/pool/lib/pool-gql-loader.service.ts [hunk 1]
```diff
@@ -458,18 +522,44 @@ export class PoolGqlLoaderService {
...
                 items: [
-                    ...aprItemsWithNoGroup.map((item) => ({
-                        ...item,
-                        apr: `${item.apr}`,
-                        subItems: [],
-                    })),
+                    ...aprItemsWithNoGroup.flatMap((item) => {
+                        let apr = `${item.apr}`;
+                        if (item.range) {
+                            apr = `${item.range.min} - ${item.range.max}`;
```
Comment (franzns) [inline modules/pool/lib/pool-gql-loader.service.ts]: this still needed?

File: modules/pool/lib/pool-gql-loader.service.ts [hunk 2]
```diff
@@ -417,12 +417,76 @@ export class PoolGqlLoaderService {
...
+                        maxThirdPartyApr += maxApr;
+                    }
+                    case 'SWAP_FEE': {
+                        swapFeeApr += maxApr;
+                        break;
+                    }
+                }
+            }
+            swapApr = `${swapFeeApr}`;
+            totalApr = `${maxTotalApr}`;
```
Comment (franzns) [inline modules/pool/lib/pool-gql-loader.service.ts:474]: bullish

**Pull request merged**